tests/src/test/java/alluxio/collections/IndexedSetConcurrencyTest.java

```diff
@@ -76,6 +78,25 @@ public long runSingleTask() {
     }
   }
 
+  private class ConcurrentAddWithCheck extends ConcurrentTask {
+    @Override
+    public long runSingleTask() {
+      long result = 0;
+      int size = ThreadLocalRandom.current().nextInt(0, MAX_SIZE);
+
+      for (int i = ThreadLocalRandom.current().nextInt(1, MAX_REPEAT_TIMES + 1); i > 0; i--) {
+        TestInfo instance = new TestInfo(size);
+        result += (mIndexedSet.add(instance) ? 1 : 0);
+        Assert.assertTrue(mIndexedSet.contains(mIdIndex, instance.getId()));
```

Comment:
We cannot run a concurrent remove at the same time of add checking. Or we would not be able to make sure wether this element should exist.